packages/@react-spectrum/dialog/docs/Dialog.mdx

```diff
@@ -43,52 +47,306 @@ import {Heading, Text} from '@react-spectrum/typography';
 
-*If the component displays or allows a user to input a value, include this section.*
+```tsx example
+<DialogTrigger>
+  <ActionButton>Upload</ActionButton>
+  {(close) => (
+    <Dialog>
+      <Image slot="hero" src="https://i.imgur.com/Z7AzH2c.png" objectFit="cover" />
+      <Header><Heading>Upload file</Heading></Header>
+      <Divider />
+      <Content>Are you sure you want to upload this file?</Content>
+      <ButtonGroup>
+        <Button variant="secondary" onPress={close}>Cancel</Button>
+        <Button variant="cta" onPress={close} autoFocus>Confirm</Button>
+      </ButtonGroup>
+    </Dialog>
+  )}
+</DialogTrigger>
+```
 
 ## Labeling
 
-*If the component supports a label prop, include this section. Please include a note about how to internationalize the content.*
+### Accessibility
+
+The title of a Dialog is typically provided via its header. By default, the Dialog sets its 
+`aria-labelledby` to match the header id, but this can be overridden by providing a `aria-labelledby` prop
+to the Dialog directly. If a visible label isn't specified, an `aria-label` must be provided instead.
+
+```tsx example
+<DialogTrigger>
+  <ActionButton>See Terms and Conditions</ActionButton>
+  {(close) => (
+    <Dialog aria-labelledby="heading-id">
+      <Text slot="header" id="heading-id">Terms and Conditions</Text> 
+      <Divider />
+      <Content>
+        <Text>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Mi proin sed libero enim. Mattis ullamcorper velit sed ullamcorper morbi tincidunt. Sed enim ut sem viverra aliquet eget sit amet tellus. Diam quis enim lobortis scelerisque fermentum dui faucibus in ornare. Diam quam nulla porttitor massa id. Eleifend mi in nulla posuere sollicitudin. Turpis nunc eget lorem dolor sed viverra ipsum nunc. Faucibus in ornare quam viverra. Risus commodo viverra maecenas accumsan lacus vel facilisis volutpat est. Nam libero justo laoreet sit amet cursus sit. Netus et malesuada fames ac. Dictum fusce ut placerat orci nulla pellentesque dignissim enim sit. Eros donec ac odio tempor orci. Ut etiam sit amet nisl purus in mollis nunc. Nisl rhoncus mattis rhoncus urna neque viverra. Convallis aenean et tortor at risus. Diam phasellus vestibulum lorem sed risus ultricies.</Text>
+      </Content>
+      <ButtonGroup>
+        <Button autoFocus variant="cta" onPress={close}>Acknowledge</Button>
+      </ButtonGroup>
+    </Dialog>
+  )}
+</DialogTrigger>
+```
 
 ## Events
 
-*If the component supports event props, include this section. Only cover the events that are important to the main functionality of the component.*
+For Dialogs, user defined callbacks should be chained with the DialogTrigger's `close` function in the press/click handler
+of the Dialog's action buttons. The following example alerts if the Dialog's save or cancel is clicked.
```

Comment:
add the word 'button' ?
-> ... if the Dialog's save or cancel button is clicked.